Source/WebKitLegacy/WebKitLegacy.xcodeproj/project.pbxproj

```diff
@@ -3184,7 +3184,7 @@
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/sh;
-			shellScript = "set -e\n\nEXP_GENERATED=\"${SCRIPT_OUTPUT_FILE_0}\"\n\n[[ -z \"${EXP_GENERATED}\" ]] && exit 1\n\nif [[ \"${WK_PLATFORM_NAME}\" != \"macosx\" ]]; then\n    EXP_WEBKIT_PLATFORM=\"${SCRIPT_INPUT_FILE_2}\"\nelse\n    EXP_WEBKIT_PLATFORM=\"${SCRIPT_INPUT_FILE_1}\"\nfi\n\ncat \"${SCRIPT_INPUT_FILE_0}\" > \"${EXP_GENERATED}\"\ncat \"${EXP_WEBKIT_PLATFORM}\" >> \"${EXP_GENERATED}\"\n\nif [ \"${WK_PLATFORM_NAME}\" != \"macosx\" ]; then\n    echo >> \"${EXP_GENERATED}\"\n    echo \"# Reexported from migrated WebCore headers\" >> \"${EXP_GENERATED}\"\n    cat \"${SCRIPT_INPUT_FILE_3}\" >> \"${EXP_GENERATED}\"\nfi\n\nif [[ \"${WK_BUILD_FOR_TESTING}\" == \"YES\" ]]; then\n    cat \"${SCRIPT_INPUT_FILE_4}\" >> \"${EXP_GENERATED}\"\nfi\n";
+			shellScript = "set -e\nexec > \"${SCRIPT_OUTPUT_FILE_0}\"\n\ncat \"${SCRIPT_INPUT_FILE_0}\"\n\nif [[ \"${WK_PLATFORM_NAME}\" == \"macosx\" ]]; then\n    cat \"${SCRIPT_INPUT_FILE_1}\"\nelse\n    cat \"${SCRIPT_INPUT_FILE_2}\"\nfi\n\necho\necho \"# Reexported from migrated WebCore headers\"\ncat \"${SCRIPT_INPUT_FILE_3}\"\n\nif [[ \"${WK_BUILD_FOR_TESTING}\" == \"YES\" ]]; then\n    cat \"${SCRIPT_INPUT_FILE_4}\"\nfi\n";
```

Comment:
Diff of the script itself:

```diff
--- A/app.kaleidoscope.kaleidoscope_dynamic_file_4BF6009F-2A93-4A82-812A-3DBB6F39D537-94887-000005E84ECEF7ED.txt
+++ B/app.kaleidoscope.kaleidoscope_dynamic_file_E99C066C-6D3B-41E8-99D3-92246F71685C-94887-000005E85C3BB80B.txt
@@ -1,24 +1,18 @@
 set -e
+exec > "${SCRIPT_OUTPUT_FILE_0}"
 
+cat "${SCRIPT_INPUT_FILE_0}"
-EXP_GENERATED="${SCRIPT_OUTPUT_FILE_0}"
 
+if [[ "${WK_PLATFORM_NAME}" == "macosx" ]]; then
+    cat "${SCRIPT_INPUT_FILE_1}"
-[[ -z "${EXP_GENERATED}" ]] && exit 1
-
-if [[ "${WK_PLATFORM_NAME}" != "macosx" ]]; then
-    EXP_WEBKIT_PLATFORM="${SCRIPT_INPUT_FILE_2}"
 else
+    cat "${SCRIPT_INPUT_FILE_2}"
-    EXP_WEBKIT_PLATFORM="${SCRIPT_INPUT_FILE_1}"
 fi
 
+echo
+echo "# Reexported from migrated WebCore headers"
+cat "${SCRIPT_INPUT_FILE_3}"
-cat "${SCRIPT_INPUT_FILE_0}" > "${EXP_GENERATED}"
-cat "${EXP_WEBKIT_PLATFORM}" >> "${EXP_GENERATED}"
-
-if [ "${WK_PLATFORM_NAME}" != "macosx" ]; then
-    echo >> "${EXP_GENERATED}"
-    echo "# Reexported from migrated WebCore headers" >> "${EXP_GENERATED}"
-    cat "${SCRIPT_INPUT_FILE_3}" >> "${EXP_GENERATED}"
-fi
 
 if [[ "${WK_BUILD_FOR_TESTING}" == "YES" ]]; then
+    cat "${SCRIPT_INPUT_FILE_4}"
-    cat "${SCRIPT_INPUT_FILE_4}" >> "${EXP_GENERATED}"
 fi
```

I refactored it a bit to avoid having to put shell redirections on every line.